src/form-elements/FormElementHTML.tsx

```diff
@@ -2,24 +2,40 @@ import * as React from 'react'
 import sanitizeHtml from '../services/sanitize-html'
 import { FormTypes } from '@oneblink/types'
 import { RepeatableSetIndexContext } from './FormElementRepeatableSet'
+import useFormSubmissionModel from '../hooks/useFormSubmissionModelContext'
+import { submissionService } from '@oneblink/sdk-core'
+import { localisationService } from '@oneblink/apps'
 
 type Props = {
   element: FormTypes.HtmlElement
 }
 
 function FormElementHTML({ element }: Props) {
+  const { formSubmissionModel, elements } = useFormSubmissionModel()
   const index = React.useContext(RepeatableSetIndexContext)
   const html = React.useMemo(() => {
-    return element.defaultValue.replace('{INDEX}', (index + 1).toString())
-  }, [element.defaultValue, index])
+    let html = element.defaultValue
+    html = html.replace('{INDEX}', (index + 1).toString())
+
+    html = submissionService.replaceElementValues(html, {
+      submission: formSubmissionModel,
+      formElements: elements,
+      formatCurrency: localisationService.formatCurrency,
+      formatDate: (v) => localisationService.formatDate(new Date(v)),
+      formatNumber: localisationService.formatNumber,
+      formatTime: (v) => localisationService.formatTime(new Date(v)),
+    })
```

Comment:
I'm pretty sure this won't handle conditionally hidden elements, I don't know how fast the function is to clean the submission model, but if it's not that performant in monkey makers land we use the conditionally shown elements to determine if an element is hidden/visible.

Reply:
Are you saying it will be slow, or it won't work?

Reply:
I'm saying the clean the submission model might be slow, but I don't know what it does under the hood.

Reply:
And I'm saying that this implementation won't handle conditionally hidden elements

Reply:
I think it does, I'll double check on it though

Reply:
Confirmed working.

Reply:
🎉